sys/net/routing/nhdp/iib_table.h

```diff
@@ -62,6 +74,8 @@ typedef struct iib_two_hop_set_entry_t {
     struct iib_link_set_entry_t *ls_elt;        /**< Pointer to corresponding link tuple */
     nhdp_addr_t *th_nb_addr;                    /**< Address of symmetric 2-hop neighbor */
     timex_t exp_time;                           /**< Time at which entry expires */
+    uint32_t metric_in;                         /**< Metric value for incoming link */
+    uint32_t metric_out;                        /**< Metric value for outgoing link */
```

Comment:
The RFC of OLSRv2 states
   >Link metrics are reported in messages using a compressed
   representation that occupies 12 bits, consisting of a 4-bit field and
   an 8-bit field.  The compressed representation represents positive
   integer values with a minimum value of 1 and a maximum value that is
   slightly smaller than the maximum 24-bit value.  Only those values
   that have exact representation in the compressed form are used.
   Route metrics are the summation of no more than 256 link metric
   values and can therefore be represented using no more than 32 bits.
   
   >Link and route metrics used in the Information Bases defined in this
   specification refer to the uncompressed values, and arithmetic
   involving them does likewise and assumes full precision in the
   result.  (How an implementation records the values is not part of
   this specification, as long as it behaves as if recording
   uncompressed values.  An implementation can, for example, use 32-bit
   values for all link and route metrics.)

An uncompressed metric value therefore needs up to 24 bits.